platforms/jvm/platform-jvm/src/main/java/org/gradle/jvm/tasks/Jar.java

```diff
@@ -64,6 +67,25 @@ public Jar() {
         metaInf = (CopySpecInternal) getRootSpec().addFirst().into("META-INF");
         metaInf.addChild().from(manifestFileTree());
         getMainSpec().appendCachingSafeCopyAction(new ExcludeManifestAction());
+        declareGracefulDegradationCondition();
+    }
+
+    private void declareGracefulDegradationCondition() {
+        if (!isExecutionTimeAccess()) {
+            // only  access project if it is still time to request graceful degradation
+            ProjectInternal project = (ProjectInternal) getProject();
+            ConfigurationCacheDegradationController degradationController = project.getServices().get(ConfigurationCacheDegradationController.class);
+            degradationController.requireConfigurationCacheDegradation(this, usingCustomCharsetDegradationReason());
+        }
+    }
+
+    private Provider<String> usingCustomCharsetDegradationReason() {
+        return Providers.changing(() -> {
+            String charset = getManifestContentCharset();
+            return charset.equals(DefaultManifest.DEFAULT_CONTENT_CHARSET)
+                ? null
+                : "Custom charset " + charset + " was used. Only " + DefaultManifest.DEFAULT_CONTENT_CHARSET + " is supported for CC";
```

Comment:
🤔 This message will be printed in the CC report, I think we should used `Configuration Cache` instead of `CC` since we're presenting this to the users